orttraining/orttraining/training_api/include/utils.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/graph/model.h"
+#include "core/session/inference_session.h"
+#include "core/providers/cpu/cpu_execution_provider.h"
+namespace onnxruntime {
+namespace training {
+namespace api {
+namespace utils {
+
+
+
+// Get names of graph inputs and outputs
+void GetGraphInputOutputNames(const std::unique_ptr<onnxruntime::InferenceSession>& session_object,
+                              std::vector<std::string>& input_names,
+                              std::vector<std::string>& output_names);
+// Fetch the parameter name from suffix: name = param_name+suffix,
+// returns True if suffix is present in name else False
+bool GetParamNameFromSuffix(const std::string& name, const std::string& suffix, std::string& param_name);
+
+// Fetch the parameter name from all possible gradient suffix: name = param_name+suffix
+// returns True if suffix is present in name else False
+bool GetParamNameFromGradient(const std::string& grad_name, std::string& param_name);
+
+// Allocate OrtValue like the input ortvalue on the same device
+Status OrtValueLike(const SessionState& sess_state, const OrtValue& input_val, OrtValue& output_val);
+
+// Create OrtValue from a single value of type T
+template <typename T>
+void WarpInOrtValue(T value,
+                    OrtValue* p_ortvalue,
+                    AllocatorPtr alloc = nullptr) {
```

Comment:
what is the advantage in making this optional? Can we make this required so that this util does not have to create a cpu provider? The caller can worry about getting the allocator...

Reply:
will do it in the next PR that deals with CUDA/CPU EP data transfer